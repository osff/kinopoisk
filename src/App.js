import React, { Component } from 'react';
import { StyleSheet, css } from 'aphrodite';
import logo from './logo.svg';
import './App.css';

class MovieListItemRatingDropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {value: this.props.value};

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }

  render() {
    return (
      <select value={this.state.value} onChange={this.handleChange}>
       <option value="1">1</option>
       <option value="2">2</option>
       <option value="3">3</option>
       <option value="4">4</option>
       <option value="5">5</option>
     </select>
    );
  }
}

class MovieListItem extends Component {
  render() {
    const actors = this.props.actors.join(", ");
    return (
      <div className={css(styles.movie_wrapper)}>
        <div>
          <img className={css(styles.movie_image)}
            src={this.props.image}
            alt={this.props.name} />
        </div>
        <div className={css(styles.movie_description)}>
          <h1>{this.props.name} ({this.props.year})</h1>
          Actors: {actors} <br />
          Rating: <MovieListItemRatingDropdown value={this.props.rating} />
        </div>

      </div>
    );
  }
}

class MovieList extends Component {
  render() {
    const moviesListItems = this.props.movies.map((movie) =>
      <MovieListItem
        key={movie.id}
        image={movie.image}
        name={movie.name}
        year={movie.year}
        actors={movie.actors}
        rating={movie.rating}
      />
    );
    return (
      <div className="movie_list">
        {moviesListItems}
      </div>
    );
  }
}

class App extends Component {
  render() {
    const movies = [
      { id: 1, image: "https://st.kp.yandex.net/images/film_iphone/iphone360_342.jpg", year: "1994", name: "Pulp fiction", actors: ['John Travolta', 'Samuel L. Jackson',], rating: 5 },
      { id: 2, image: "https://st.kp.yandex.net/images/film_iphone/iphone360_326.jpg", year: "1994", name: "The Shawshank Redemption", actors: ['Tim Robbins', 'Morgan Freeman',], rating: 4 },
      { id: 3, image: "https://st.kp.yandex.net/images/film_iphone/iphone360_435.jpg", year: "1999", name: "The Green Mile", actors: ['Tom Henks', 'David Mors',], rating: 3 },
      { id: 4, image: "https://st.kp.yandex.net/images/film_iphone/iphone360_448.jpg", year: "1994", name: "Forrest Gump", actors: ['Tom Henks', 'Robin Rite',], rating: 2 },
      { id: 5, image: "https://st.kp.yandex.net/images/film_iphone/iphone360_329.jpg", year: "1993", name: "Schindler's List", actors: ['Liam Nisson', 'Ben Ringslie',], rating: 1 },
    ]
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Kinopoisk</h2>
        </div>
        <MovieList movies={movies} />
      </div>
    );
  }
}

export default App;
export {MovieListItemRatingDropdown};

const styles = StyleSheet.create({
    movie_wrapper: {
        display: 'flex',
        justifyContent: 'center',
        width: "80vw",
        margin: "1em auto"
    },

    movie_description: {
        padding: '1em 0 0 2em',
        width: "50%"
    },

    movie_image: {
      width: "70%"
    }
});
