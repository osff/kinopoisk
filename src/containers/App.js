import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import logo from '../logo.svg';
import '../App.css';
import '../index.css';
import SaveButton from '../components/SaveButton'
import FilteredMovieList from '../containers/FilteredMovieList'
import FilteredPanel from '../containers/FilteredPanel'
import { changeRating } from '../actions'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {updatedRating: []};
    this.handleRatingChange = this.handleRatingChange.bind(this);
    this.handleSaveClick = this.handleSaveClick.bind(this);
    this.onFilterChange = this.onFilterChange.bind(this);
  }

  _updateRatingInArrayOfMovies(arr, data) {
    let {id, rating} = data;
    let arr_copy = arr.slice();
    let foundExisted = false;
    arr_copy.some((ratingChangeItem) => {
      if (ratingChangeItem['id'] === id) {
        ratingChangeItem['rating'] = rating;
        foundExisted = true;
        return true;
      }
      return false;
    });
    if (! foundExisted) {
      arr_copy.push({
        'id': id,
        'rating': rating
      });
    }
    return arr_copy;
  }

  handleRatingChange(data) {
    this.props.changeRating(data);
    this.setState({
      updatedRating: this._updateRatingInArrayOfMovies(this.state.updatedRating, data),
    });
  }

  handleSaveClick() {
    alert(JSON.stringify(this.state.updatedRating));
  }

  onFilterChange(value) {
    alert(value);
  }

  render() {
    // const filterBy = Object.keys(this.props.movies[0]);
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Kinopoisk</h2>
        </div>
        <div style={styles.app_body}>
          <FilteredPanel />
          <FilteredMovieList onRatingChange={this.handleRatingChange} />
          <SaveButton onClick={this.handleSaveClick} />
        </div>
      </div>
    )
  }
}

App.PropTypes = {
  movies: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  movies: state.moviesReducer
})

const mapDispatchToProps = {
  changeRating: changeRating
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

const styles = {
  app_body: {
    width: "80%",
    margin: "0 auto"
  }
}
