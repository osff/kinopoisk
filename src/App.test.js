import React from 'react';
import ReactDOM from 'react-dom';
// import {App, MovieListItemRatingDropdown} from './App';
import App from './App';
// import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const doc = document.createElement('document');
  const body = document.createElement('body');
  const div = document.createElement('div');
  doc.appendChild(body);
  body.appendChild(div);
  ReactDOM.render(<App />, div);
});

// test('Link changes the class when hovered', () => {
//   const component = renderer.create(
//     <MovieListItemRatingDropdown value="4" />
//   );
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
//
//   // manually change value
//   tree.props.value = 5;
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
//
//   // manually change value
//   tree.props.value = 1;
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
