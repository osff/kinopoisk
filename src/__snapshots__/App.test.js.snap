// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<select
  onChange={[Function]}
  value="4"
>
  <option
    value="1"
  >
    1
  </option>
  <option
    value="2"
  >
    2
  </option>
  <option
    value="3"
  >
    3
  </option>
  <option
    value="4"
  >
    4
  </option>
  <option
    value="5"
  >
    5
  </option>
</select>
`;

exports[`Link changes the class when hovered 2`] = `
<select
  onChange={[Function]}
  value="4"
>
  <option
    value="1"
  >
    1
  </option>
  <option
    value="2"
  >
    2
  </option>
  <option
    value="3"
  >
    3
  </option>
  <option
    value="4"
  >
    4
  </option>
  <option
    value="5"
  >
    5
  </option>
</select>
`;

exports[`Link changes the class when hovered 3`] = `
<select
  onChange={[Function]}
  value="4"
>
  <option
    value="1"
  >
    1
  </option>
  <option
    value="2"
  >
    2
  </option>
  <option
    value="3"
  >
    3
  </option>
  <option
    value="4"
  >
    4
  </option>
  <option
    value="5"
  >
    5
  </option>
</select>
`;
